string source = rogr
string target = waterpump
time timestamp
bool pump_on

